packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -28,4 +47,53 @@ export class Snapshot {
   async delete() {
     // TODO: CDX-359: Delete snapshot once previewed
   }
+
+  get lastestReport(): ResourceSnapshotsReportModel {
```

Comment:
By refactoring it so `validate()` control the whole flow, instead of 2 method, this check would go away.